pkg/process/metadata/workloadmeta/collector/process.go

```diff
@@ -0,0 +1,135 @@
+	}
+}
+
+type collector struct {
+	ddConfig config.ConfigReader
+
+	processData *checks.ProcessData
+
+	wlmExtractor *workloadmetaExtractor.WorkloadMetaExtractor
+	grpcServer   *workloadmetaExtractor.GRPCServer
+
+	pidToCid map[int]string
+
+	collectionClock clock.Clock
+}
+
+func (c *collector) Start(ctx context.Context, store workloadmeta.Store) error {
+	err := c.grpcServer.Start()
+	if err != nil {
+		return err
+	}
+
+	collectionTicker := c.collectionClock.Ticker(
+		c.ddConfig.GetDuration("workloadmeta.local_process_collector.collection_interval"),
+	)
+
+	filter := workloadmeta.NewFilter([]workloadmeta.Kind{workloadmeta.KindContainer}, workloadmeta.SourceAll, workloadmeta.EventTypeAll)
+	containerEvt := store.Subscribe(collectorId, workloadmeta.NormalPriority, filter)
+
+	go c.run(ctx, store, containerEvt, collectionTicker)
+
+	return nil
+}
+
+func (c *collector) run(ctx context.Context, store workloadmeta.Store, containerEvt chan workloadmeta.EventBundle, collectionTicker *clock.Ticker) {
+	defer c.grpcServer.Stop()
+	defer store.Unsubscribe(containerEvt)
+	defer collectionTicker.Stop()
+
+	log.Info("Starting local process collector")
+
+	for {
+		select {
+		case evt := <-containerEvt:
+			c.handleContainerEvent(evt)
+		case <-collectionTicker.C:
+			err := c.processData.Fetch()
+			if err != nil {
+				log.Error("Error fetching process data:", err)
+			}
+		case <-ctx.Done():
+			log.Infof("The %s collector has stopped", collectorId)
+			return
+		}
+	}
+}
+
+// Pull is unused at the moment used due to the short frequency in which it is called.
+// In the future, we should use it to poll for processes that have been collected and store them in workload-meta.
+func (c *collector) Pull(_ context.Context) error {
+	return nil
+}
+
+func (c *collector) handleContainerEvent(evt workloadmeta.EventBundle) {
+	defer close(evt.Ch)
+
+	for _, evt := range evt.Events {
+		ent := evt.Entity.(*workloadmeta.Container)
+		switch evt.Type {
+		case workloadmeta.EventTypeSet:
+			// Should be safe, even on windows because PID 0 is the idle process and therefore must always belong to the host
+			if ent.PID != 0 {
+				c.pidToCid[ent.PID] = ent.ID
+			}
+		case workloadmeta.EventTypeUnset:
+			delete(c.pidToCid, ent.PID)
+		}
+	}
+
+	c.wlmExtractor.SetLastPidToCid(c.pidToCid)
+}
+
+// Enabled checks to see if we should enable the local process collector.
+// Since it's job is to collect processes when the process check is disabled, we only enable it when `process_config.process_collection.enabled` == false
+// Additionally, if the remote process collector is not enabled in the core agent, there is no reason to collect processes. Therefore, we check `workloadmeta.remote_process_collector.enabled`
+// Finally, we only want to run this collector in the process agent, so if we're running as anything else we should disable the collector.
+func Enabled(cfg config.ConfigReader) bool {
+	if flavor.GetFlavor() != flavor.ProcessAgent {
```

Comment:
Do we still need to use flavor ? I think this code is only executed in the process agent

Reply:
Ah that's a good point. I removed it.